include/oneapi/dpl/internal/random_impl/discard_block_engine.h

```diff
@@ -82,25 +82,42 @@ class discard_block_engine
         if (!__num_to_skip)
             return;
 
-        for (; __num_to_skip > 0; --__num_to_skip)
-            generate_internal_scalar<internal::type_traits_t<result_type>::num_elems>();
+        if (__num_to_skip < (used_block - n_))
+        {
+            n_ += __num_to_skip;
```

Comment:
unfortunately it is a gap in C++ RNG design I think
n_ < used_block - it is a rule 
current code-branch is for __num_to_skip < (used_block - n_) condition
so that overflow can happen only in the specific cases but anyway in these cases base rule will not work